---
title: "S2 pioneer factor MS"
author: "Tyler Gibson"
date: "8/24/2021"
output: html_document
---

# Abstract

# Introduction

# Results

## Ectopically expressed pioneer factors bind and open closed chromatin

-   Gain of function system is necessary to understand mechanisms of PF activity

-   Establish system for ectopic expression of pioneer factors

-   Identification of physiological expression levels

-   Define binding classes I-III and describe characteristics (distribution, promoters/enhancers, etc.)

## Chromatin opening correlates with transcriptional activation

-   Description of RNA-seq data: number of differential genes, overlap with ChIP data, GO analysis.

-   Comparison of RNA-seq and ATAC data: chromatin opening correlates with transcriptional activation.

-   Class III promoters are most likely to be associated with upregulated genes

## Twist binds closed chromatin extensively and opens chromatin at a subset of sites

-   Introduce Twist and justify the comparison to Zld/Grh.

-   Twi binds closed chromatin and opens at some sites

-   Class I-III sites, distribution, etc.

-   Twi also promotes transcriptional activation.

-   Description of RNA-seq data: number of differential genes, overlap with ChIP data, GO analysis.

-   Comparison of RNA-seq and ATAC

## Ectopically expressed transcription factors bind opportunistically to active chromatin

-   Overlap of class I regions suggests nonspecific binding

-   Analysis of published ChIP data shows that class I regions are within active chromatin

-   RNA-seq and ATAC-seq for class I to show that the binding is not functional

## Class II and III regions are mostly in naïve chromatin

-   Analysis of chromatin at class II-III regions: mostly devoid of histone modifications

-   Class III regions have higher H3 and H1 occupancy compared to class II

-   Class II and III have poorly positioned nucleosomes

## Chromatin opening by pioneer factors is driven by motif content

-   De novo motif enrichment for ChIP peaks (supplemental)

-   Motif analysis

## Ectopic Zld, Grh and Twi fail to access many endogenous binding sites

## Repressive chromatin is a barrier to pioneer-factor binding

## Pioneer-factor binding and chromatin opening is concentration dependent

## Regions outside the DNA-binding domain are required for binding and opening of closed chromatin

# Discussion

# Materials and Methods

## Reagents

| Reagent                            | Source/vendor            | Catalog/accession \# |
|------------------------------|-----------------------|-------------------|
| pMT-puro                           | AddGene                  | 17923                |
| Gibson Cloning                     | New England Biolabs      | E2611                |
| Schneider's Medium                 | Thermo Fisher Scientific | 21720001             |
| Fetal Bovine Serum                 | Omega Scientific         | FB-02                |
| Antibiotic/antimycotic             | Thermo Fisher Scientific | 15240062             |
| Effectene Transfection Reagent     | Qiagen                   | 301427               |
| PVDF Membrane                      |                          |                      |
| Formaldehyde                       |                          |                      |
| **Antibodies**                     |                          |                      |
|                                    |                          |                      |
| Anti-Zld                           | Harrison et al., 2010    |                      |
| Anti-Grh                           | Harrison et al., 2010    |                      |
| Anti-Twi                           | He et al., 2015          |                      |
| Anti-HA-peroxidase clone 3F10      | Roche                    | 12013819001          |
| Anti-Tubulin DM1A                  | Sigma                    | T6199-200UL          |
| goat anti-rabbit IgG-HRP conjugate | Bio-Rad                  | 1706515              |
| Anti-HA clone 12CA5                |                          |                      |
| Anti-H3K27me3                      |                          |                      |
|                                    |                          |                      |
|                                    |                          |                      |

## Cloning

*zld*, *grh*, or *twi* cDNA (isoforms RB, RH, and RA, respectively) were cloned into pMT-puro plasmid (Addgene) via Gibson cloning (New England Biolabs). For cloning of the Zld and Grh DNA-binding domains, DNA encoding amino acids 1114-1487 (Zld) or amino acids 603-1032 (Grh) was cloned into pMT-puro.

-   **TG2A- Zld RB**

-   **TG3A- Grh PH**

-   **TG4A**

-   **TG6A**

-   **TG8A**

-   **MF8A**

-   GFP-DBD lines

## Cell culture and generation of stable cell lines

Drosophila S2 cells were cultured at 27°C in Schneider's medium (Thermo Fisher Scientific) supplemented with 10% fetal bovine serum (Omega Scientific) and antibiotic/antimycotic (Thermo Fisher Scientific). For the generation of stable cell lines, cells were plated at 5✕10^5^ cells per mL. After 24 hours, cells were transfected with 10 µg plasmid DNA using Effectene transfection reagent (Qiagen). After an additional 24 hours, puromycin was added to a final concentration of 2 µg/mL. Stable cell lines were recovered after 2-3 weeks of selection. Following recovery of stable cell lines, cells were cultured with 1 µg/mL puromycin.

## Induction of transcription factor expression

Transcription factor expression was induced by adding CuSO~4~ to the cell culture media. Expression of protein at physiological levels was achieved with 1000, 100, or 40 µM CuSO~4~ for Zld, Grh or Twi, respectively (Supplemental figure X). For experiments with sub- or supraphysiological protein levels (Fig. 5), the following CuSO~4~ concentrations were used: 500 µM (Zld subphysiological), 1000 µM (Zld physiological) or 1500 µM (Zld supraphysiological); 25 µM (Grh subphysiological), 100 µM (Grh physiological) or 400 µM (Grh supraphysiological); 10 µM (Twi subphysiological), 40 µM (Twi physiological) or 160 µM (Twi supraphysiological). For all induction experiments, cells were plated at 1✕10^6^ cells per mL. Unless otherwise specified, cells were harvested 48 hours after induction.

## Immunoblotting

Proteins were separated on denaturing polyacramide gels before transfer to a 0.45 μm Immobilon-P PVDF membrane (Millipore) in transfer buffer (25 mM Tris, 200 mM Glycine, 20% methanol) for 60 min (75 min for Zld) at 500mA at 4°C. The membranes were blocked with BLOTTO (2.5% non-fat dry milk, 0.5% BSA, 0.5% NP-40, in TBST) for 30 min at room temperature and then incubated with anti-Zld (1:750) (Harrison et al., 2010), anti-Grh (1:1000) (Harrison et al., 2010), anti-Twi (1:1000) (He et al. 2015), anti-HA-peroxidase (1:500) (clone 3F10, Roche), or anti-Tubulin (DM1A, 1:5000) (Sigma), overnight at 4°C. The secondary incubation was performed with goat anti-rabbit IgG-HRP conjugate (1:3000) (Bio-Rad) for 1 hr at room temperature. Blots were treated with SuperSignal West Pico PLUS chemiluminescent substrate (Thermo Fisher Scientific) and visualized using the Azure Biosystems c600 or Kodak/Carestream BioMax Film (VWR).

## Immunostaining

## ChIP-seq

For each ChIP experiment, \~50 million cells were crosslinked by adding methanol-free formaldehyde directly to the cell culture medium to a final concentration of 0.8%. Cells were rotated on a nutator for 7 minutes at room temperature (RT) before quenching of crosslinking by adding glycine to XX mM. Crosslinked cells were pelleted by centrifugation at 600✕g for 3 minutes. Cells were washed twice with 1X PBS.

## ChIP-seq analysis

## RNA-seq

## RNA-seq analysis

## ATAC-seq

## ATAC-seq analysis

## CUT&RUN

## CUT&RUN analysis

## Treatment of S2 cells with Tazemetostat

## Analysis of previously published data

## Data and code availability

# Figure Legends

**Figure 1: Ectopic pioneer factors open chromatin and activate transcription.**

**Extended data Figure 1: Stable cell lines allow inducible expression of transcription factors at physiological concentrations.**

**Figure 2: Twist binds closed chromatin extensively and drives accessibility at a limited number of sites.**

**Figure 3: Motif content shapes pioneer-factor activity**

**Figure 4: H3K27me3 and H3K9me3 are barriers to pioneer-factor binding.**

**Figure 5: Binding and opening of closed chromatin is concentration dependent.**

**Figure 6: Regions outside the DNA-binding domain are required for pioneer-factor function.**

**Figure 7: Pioneer-factor function requires stable chromatin occupancy.**

**Information to include in supplemental figures:**

-   Schematic of stable cell line generation.

-   Cell line validation by RNA-seq

-   Verification of protein expression.

-   Determination of physiological protein levels.

-   QC for genomics datasets: e.g. PCA, replicate correlation, example genome browsers of individual replicates.

-   Antibody controls: IgG, IPs in WT cells, etc.

-   CUT&RUN time course for Zld and Grh?

-   De novo motif results

-   Analysis of promoter motifs

-   GO analysis for RNA-seq data

-   
