<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.189">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Defining properties regulating pioneer-factor binding and function</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>


<script src="manuscript_files/libs/clipboard/clipboard.min.js"></script>
<script src="manuscript_files/libs/quarto-html/quarto.js"></script>
<script src="manuscript_files/libs/quarto-html/popper.min.js"></script>
<script src="manuscript_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="manuscript_files/libs/quarto-html/anchor.min.js"></script>
<link href="manuscript_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="manuscript_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="manuscript_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="manuscript_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="manuscript_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Defining properties regulating pioneer-factor binding and function</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="abstract" class="level1">
<h1>Abstract</h1>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<section id="ectopically-expressed-pioneer-factors-bind-and-open-closed-chromatin" class="level2">
<h2 class="anchored" data-anchor-id="ectopically-expressed-pioneer-factors-bind-and-open-closed-chromatin">Ectopically expressed pioneer factors bind and open closed chromatin</h2>
<ul>
<li><p>Gain of function system is necessary to understand mechanisms of PF activity</p></li>
<li><p><del>Establish system for ectopic expression of pioneer factors</del></p></li>
<li><p><del>Identification of physiological expression levels</del></p></li>
<li><p><del>Define binding classes I-III and describe characteristics (distribution, promoters/enhancers, etc.)</del></p></li>
<li><p>Time course and dynamics of binding to different classes (supplement)</p></li>
</ul>
<p>To investigate the mechanisms of chromatin binding and opening by pioneer factors, we generated stable cell lines inducibly expressing Zelda (Zld) and Grainy head (Grh) in Drosophila S2 cells (Extended Data Fig. XX), where neither factor is endogenously expressed. We validated our wild-type (WT) S2 cells by performing RNA-seq and verifying the expression of known S2 cell markers (Extended Data Fig. 1b) <span class="citation" data-cites="Cherbas2011">(<a href="#ref-Cherbas2011" role="doc-biblioref">Cherbas et al. 2010</a>)</span>. We confirmed that Zld and Grh RNA (Extended Data Fig. XX) and protein (Extended Data Fig. XX) were not present in WT S2 cells. The stable cell lines express Zld or Grh under the control of the metallothionein promoter, which allows inducible and tunable expression with the addition of CuSO<sub>4.</sub> To determine protein levels that would be approximately physiological, we compared Zld or Grh levels in cells with varying concentrations of CuSO<sub>4</sub> to the levels of endogenous expression found in Drosophila embryos (Extended Data Fig. XX). Physiological protein levels were achieved at 1000 mM or 100 mM CuSO<sub>4</sub> for Zld or Grh, respectively.</p>
<p>To determine how exogenous Zld and Grh interact with preexisting chromatin in S2 cells, we expressed each factor at physiological levels and determined genome-wide binding sites using chromatin immunoprecipitation coupled with sequencing (ChIP-seq). We identified 14455 peaks for Zld and 13833 for Grh. To control for antibody specificity, we performed additional ChIP experiments in WT cells using IgG, anti-Zld, or anti-Grh (Extended Data Fig. XX). We excluded any peaks that were called in these datasets, ensuring that we analyzed peaks that were detected only upon ectopic expression of Zld or Grh. <!--# Not sure the best way to discuss the control ChIP experiments. Should I mention how many peaks we detect in WT cells? What supplemental figures should I include? Should peak calling be done using IgG or WT cells as control instead of input? --></p>
<p>Having established that Zld and Grh can bind to thousands of sites in S2 cells, we performed ATAC-seq on WT cells or cells expressing Zld or Grh to determine if these factors bind and open closed chromatin in S2 cells. We performed differential accessibility analysis comparing Zld or Grh-expressing cells to WT cells. We detected 6546 differentially accessible regions for Zld (3769 increased, 2777 decreased) and 13732 for Grh (4784 increased, 8948 decreased) (Extended Data Fig. XX). <!--# Add more information here about sites with differential accessibility. Include percentage of increased and decreased sites that are bound by Zld or Grh. Comment on how much is specific vs. secondary effect, and justify defining ChIP peak classes that include only a subset of DA sites. --></p>
<p>To characterize the extent of chromatin opening by Zld and Grh, we integrated our ChIP-seq and ATAC-seq data and identified three classes of binding sites (Fig. 1a,b). Class I sites are accessible in WT cells and remain accessible when bound by ectopically-expressed Zld or Grh. Class II sites are inaccessible in WT cells, however binding of Zld or Grh does not lead to chromatin opening. Class III sites are inaccessible in WT cells and Zld or Grh binding results in chromatin opening. The majority of ChIP-seq peaks for Zld and Grh were class I sites, with a subset of binding sites in closed chromatin, and a small subset undergoing chromatin opening (Fig. 1c,f). While class I binding sites occured mostly at promoters, the majority of class II and III sites were at promoter-distal sites (Fig. 1d,g). Genome-wide analysis of signa <!--#  -->l intensities at all class II and III sites confirmed that these sites lack ATAC signal in WT cells and that class III sites undergo robust chromatin opening upon Zld or Grh induction (Fig. 1e,h). Together, these data show that when expressed exogenously at physiolgical levels, Zld and Grh can bind and open closed chromatin at a subset of binding sites.</p>
<!--# Could potentially add the cut&run time course here? Think about where to put this, or if it is worth including at all. -->
</section>
<section id="chromatin-opening-correlates-with-transcriptional-activation" class="level2">
<h2 class="anchored" data-anchor-id="chromatin-opening-correlates-with-transcriptional-activation">Chromatin opening correlates with transcriptional activation</h2>
<ul>
<li><p>Description of RNA-seq data: number of differential genes, overlap with ChIP data, GO analysis.</p></li>
<li><p>Comparison of RNA-seq and ATAC data: chromatin opening correlates with transcriptional activation.</p></li>
<li><p>Class III promoters are most likely to be associated with upregulated genes</p></li>
</ul>
<p>Having established that ectopically expressed Zld and Grh bind and open closed chromatin, we sought to determine if binding of these factors also led to transcriptional activation. To identify changes in gene expression upon Zld or Grh induction, we performed RNA-seq and compared induced cells to WT cells lacking Zld or Grh expression. Differential expression analysis using DESeq2 <span class="citation" data-cites="love2014">(<a href="#ref-love2014" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span> identified XXX differentially expressed genes for Zld and XXX for Grh (Extended Data Fig. XX). Upregulated genes were significantly enriched for Zld or Grh binding, while downregulated genes are not (Extended Data Fig. XX <!--# include statistics  -->), suggesting that binding of these factors mainly leads to transcriptional activation and not repression. <!--# Consider mentioning Grh as a repressor here and giving examples of genes where Grh may be repressing. Discuss if Grh-mediated transcriptional repression coincides with chromatin opening, chromatin closing, or neither --> Gene ontology (GO) analysis indicated that the genes activated by each factor are consistent with the endogenous functions of Zld and Grh (Extended Data Fig. XX). <!--# Add more detailed information on GO analysis here --></p>
<p>To better understand the relationship between chromatin opening and transcriptional activation, we directly compared our RNA-seq and ATAC-seq data. Overall, chromatin opening was correlated with transcriptional activation (Fig. 1i-l). sites with significantly increased chromatin accessibility, but not decreased or unchanged sites, had larger RNA-seq fold change values (Extended Data Fig. XX <!--# add statistics -->). When we classified our three classes of ChIP-seq based on changes in gene expression, we found that class III sites were most likely to be proximal to upregulated genes (Fig. 1j,l). This was particularly evident at class III promoters, which were most likely to correspond to upregulated genes. Together, we conclude that while binding and chromatin opening can happen in the absence of transcriptional changes, chromatin opening is correlated with transcriptional activation.</p>
</section>
<section id="twist-binds-closed-chromatin-extensively-and-opens-chromatin-at-a-subset-of-sites" class="level2">
<h2 class="anchored" data-anchor-id="twist-binds-closed-chromatin-extensively-and-opens-chromatin-at-a-subset-of-sites">Twist binds closed chromatin extensively and opens chromatin at a subset of sites</h2>
<ul>
<li><p>Introduce Twist and justify the comparison to Zld/Grh.</p></li>
<li><p>Twi binds closed chromatin and opens at some sites.</p></li>
<li><p>Mention analysis of published Abd-A and Ubx cell lines.</p></li>
<li><p>Class I-III sites, distribution, etc.</p></li>
<li><p>Twi also promotes transcriptional activation</p></li>
<li><p>Description of RNA-seq data: number of differential genes, overlap with ChIP data, GO analysis.</p></li>
<li><p>Comparison of RNA-seq and ATAC</p></li>
<li><p>Mention differences between Twi vs.&nbsp;Zld and Grh</p></li>
</ul>
<p>Despite the differences between Zld and Grh during Drosophila development, both proteins behaved similarly when expressed in our exogenous S2 cell system. To provide additional context to our findings, we performed additional experiments with the transcription factor Twist (Twi). Although Twi is also essential for regulating Drosophila embryonic development, Twi has a number if distnict properties from Zld and Grh, making it an interesting point of comparison. Twi is a master regulator of mesodermal cell fate (REFS). Twi has many binding sites that are dynamic during early embryonic development <span class="citation" data-cites="Wilczynski2010a">(<a href="#ref-Wilczynski2010a" role="doc-biblioref">Wilczyński and Furlong 2010</a>)</span>, in contrast to Zld and Grh. Although Twi was hypothesized to be a pioneer factor <span class="citation" data-cites="Sandmann2007">(<a href="#ref-Sandmann2007" role="doc-biblioref">Sandmann et al. 2007</a>)</span>, pioneering activity by Twi had not previously been tested. In the early Drosophila embryo, mutation of Zld binding sites leads to a loss of Twi binding to the <em>cactus</em> enhancer, suggesting that pioneering by Zld may be required for Twi binding in the embryo <span class="citation" data-cites="Yáñez-Cuna2012">(<a href="#ref-Yáñez-Cuna2012" role="doc-biblioref">Yáñez-Cuna et al. 2012</a>)</span>. To test if Twi is a pioneer factor, we generated additional stable cell lines allowing inducible expression of Twi at approximately physiological levels (Extended Data Fig. XX). We characterized Twi binding and activity using ChIP-seq, ATAC-seq, and RNA-seq. Comparison of Twi ChIP-seq and ATAC-seq data allowed for the identification of the same classes of binding sites that were found for Zld and Grh (Fig 2a,b). Similar to Zld and Grh, class I binding sites were enriched at promoters, while class II and III binding sites were mostly promoter-distal (Fig 2c). Twi bound extensively to closed chromatin, with XX% of binding sites lacking detectible chromatin accessibility (Fig. 2b). At a small subset of Twi binding sites, binding to closed chromatin resulted in chromatin opening (Fig 2a-c). Analysis of RNA-seq data revealed that Twi expression resulted in transcriptional activation of hundreds of genes (Extended Data Fig. XX). Upregulated genes were significantly enriched for Twi binding relative to unchanged genes ((Extended Data Fig. XX). GO analysis demonstrated that genes upregulated upon Twi expression are enriched for mesodermal and muscle-specific genes, consistent with the endogenous function of Twi (Extended Data Fig. XX).</p>
<p>Considering the similarities in our system between Zld, Grh, and Twi, we performed additional analyses to determine if there were any notable differences between the behavior of these proteins when exogenously expressed.</p>
</section>
<section id="ectopically-expressed-transcription-factors-bind-opportunistically-to-active-chromatin" class="level2">
<h2 class="anchored" data-anchor-id="ectopically-expressed-transcription-factors-bind-opportunistically-to-active-chromatin">Ectopically expressed transcription factors bind opportunistically to active chromatin</h2>
<ul>
<li><p>Overlap of class I regions suggests nonspecific binding</p></li>
<li><p>Analysis of published ChIP data shows that class I regions are within active chromatin</p></li>
<li><p>RNA-seq and ATAC-seq for class I to show that the binding is not functional</p></li>
</ul>
</section>
<section id="class-ii-and-iii-regions-are-mostly-in-naïve-chromatin" class="level2">
<h2 class="anchored" data-anchor-id="class-ii-and-iii-regions-are-mostly-in-naïve-chromatin">Class II and III regions are mostly in naïve chromatin</h2>
<ul>
<li><p>Analysis of chromatin at class II-III regions: mostly devoid of histone modifications</p></li>
<li><p>Class III regions have higher H3 and H1 occupancy compared to class II</p></li>
</ul>
</section>
<section id="chromatin-opening-by-pioneer-factors-is-driven-by-motif-content" class="level2">
<h2 class="anchored" data-anchor-id="chromatin-opening-by-pioneer-factors-is-driven-by-motif-content">Chromatin opening by pioneer factors is driven by motif content</h2>
<ul>
<li><p>De novo motif enrichment for ChIP peaks (supplemental)</p></li>
<li><p>Motif analysis: percent with motif, average n motifs, motif quality</p></li>
<li><p>Motif positioning around nucleosomes? (supplemental)</p></li>
</ul>
</section>
<section id="ectopic-zld-grh-and-twi-fail-to-access-many-endogenous-binding-sites" class="level2">
<h2 class="anchored" data-anchor-id="ectopic-zld-grh-and-twi-fail-to-access-many-endogenous-binding-sites">Ectopic Zld, Grh and Twi fail to access many endogenous binding sites</h2>
<ul>
<li><p>Zelda binding is more similar to the brain than the embryo</p></li>
<li><p>Grh binding is stable across tissues</p></li>
<li><p>Twi sites in S2 cells have low level binding in embryo, but strongest peaks from embryo are not bound</p></li>
</ul>
</section>
<section id="repressive-chromatin-is-a-barrier-to-pioneer-factor-binding" class="level2">
<h2 class="anchored" data-anchor-id="repressive-chromatin-is-a-barrier-to-pioneer-factor-binding">Repressive chromatin is a barrier to pioneer-factor binding</h2>
<ul>
<li><p>Enrichment of repressive marks at sites bound in other tissues</p></li>
<li><p>Results of Taz experiments (TBD)</p></li>
<li><p>Comment on sites that are not bound and lack repressive chromatin.</p></li>
</ul>
</section>
<section id="pioneer-factor-binding-and-chromatin-opening-is-concentration-dependent" class="level2">
<h2 class="anchored" data-anchor-id="pioneer-factor-binding-and-chromatin-opening-is-concentration-dependent">Pioneer-factor binding and chromatin opening is concentration dependent</h2>
<ul>
<li><p>For all three factors, increased concentration results in increased binding and opening of class II and III sites.</p></li>
<li><p>For all three factors, increased concentration leads to novel binding sites.</p></li>
<li><p>For Zld and Twi, increased concentration is insufficient to allow binding to repressed sites.</p></li>
<li><p>For Grh, increased concentration does allow binding and opening at repressed sites</p></li>
</ul>
</section>
<section id="regions-outside-the-dna-binding-domain-are-required-for-binding-and-opening-of-closed-chromatin" class="level2">
<h2 class="anchored" data-anchor-id="regions-outside-the-dna-binding-domain-are-required-for-binding-and-opening-of-closed-chromatin">Regions outside the DNA-binding domain are required for binding and opening of closed chromatin</h2>
<ul>
<li><p>Generation of stable cell lines to express DBDs at same concentration as FL proteins</p></li>
<li><p>Zld DBD is totally unable to bind to closed sites.</p></li>
<li><p>In supplement, show Zld DBD binding specifically at class II and III sites with CAGGTAG to support that this is not due to a loss of ZnF2</p></li>
<li><p>Grh DBD has reduced ability to access closed chromatin</p></li>
<li><p>Neither DBD is sufficient for chromatin opening.</p></li>
</ul>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
</section>
<section id="materials-and-methods" class="level1">
<h1>Materials and Methods</h1>
<section id="reagents" class="level2">
<h2 class="anchored" data-anchor-id="reagents">Reagents</h2>
<table class="table">
<colgroup>
<col style="width: 26%">
<col style="width: 24%">
<col style="width: 24%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th>Reagent</th>
<th>Source/vendor</th>
<th>Catalog/accession #</th>
<th>Notes</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>pMT-puro</td>
<td>AddGene</td>
<td>17923</td>
<td></td>
</tr>
<tr class="even">
<td>Gibson Cloning</td>
<td>New England Biolabs</td>
<td>E2611</td>
<td></td>
</tr>
<tr class="odd">
<td>Schneider’s Medium</td>
<td>Thermo Fisher Scientific</td>
<td>21720001</td>
<td></td>
</tr>
<tr class="even">
<td>Fetal Bovine Serum</td>
<td>Omega Scientific</td>
<td>FB-02</td>
<td></td>
</tr>
<tr class="odd">
<td>Antibiotic/antimycotic</td>
<td>Thermo Fisher Scientific</td>
<td>15240062</td>
<td></td>
</tr>
<tr class="even">
<td>Effectene Transfection Reagent</td>
<td>Qiagen</td>
<td>301427</td>
<td></td>
</tr>
<tr class="odd">
<td>PVDF Membrane</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Formaldehyde</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Protease inhibitor tablets</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>NEBNext Ultra II DNA library prep kit</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>TruSeq v2 RNA library prep kit</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Trizol</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td><strong>Antibodies</strong></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Anti-Zld</td>
<td>Harrison et al., 2010</td>
<td></td>
<td>ChIP: 6 µL , WB: 1:750</td>
</tr>
<tr class="odd">
<td>Anti-Grh</td>
<td>Harrison et al., 2010</td>
<td></td>
<td>ChIP: 8 µL, WB: 1:1000</td>
</tr>
<tr class="even">
<td>Anti-Twi</td>
<td>He et al., 2015</td>
<td></td>
<td>ChIP: 10 µg, WB: 1:1000</td>
</tr>
<tr class="odd">
<td>Anti-HA-peroxidase clone 3F10</td>
<td>Roche</td>
<td>12013819001</td>
<td>WB: 1:500</td>
</tr>
<tr class="even">
<td>Anti-Tubulin DM1A</td>
<td>Sigma</td>
<td>T6199-200UL</td>
<td>WB: 1:5000</td>
</tr>
<tr class="odd">
<td>goat anti-rabbit IgG-HRP conjugate</td>
<td>Bio-Rad</td>
<td>1706515</td>
<td>WB: 1:3000</td>
</tr>
<tr class="even">
<td>Anti-HA clone 12CA5</td>
<td>Roche</td>
<td></td>
<td>ChIP: 7.5 µL</td>
</tr>
<tr class="odd">
<td>Anti-H3K27me3</td>
<td>Cell Signaling Technology</td>
<td></td>
<td>WB: 1:1000, CR: 0.5 µL</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</section>
<section id="cloning" class="level2">
<h2 class="anchored" data-anchor-id="cloning">Cloning</h2>
<p><em>zld</em>, <em>grh</em>, or <em>twi</em> cDNA (isoforms RB, RH, and RA, respectively) were cloned into pMT-puro plasmid (Addgene) via Gibson cloning (New England Biolabs). For cloning of the Zld and Grh DNA-binding domains, DNA encoding amino acids 1114-1487 (Zld) or amino acids 603-1032 (Grh) was cloned into pMT-puro.</p>
<ul>
<li><p><strong>TG2A- Zld RB</strong></p></li>
<li><p><strong>TG3A- Grh PH</strong></p></li>
<li><p><strong>TG4A</strong></p></li>
<li><p><strong>TG6A</strong></p></li>
<li><p><strong>TG8A</strong></p></li>
<li><p><strong>MF8A</strong></p></li>
<li><p>GFP-DBD lines</p></li>
</ul>
</section>
<section id="cell-culture-and-generation-of-stable-cell-lines" class="level2">
<h2 class="anchored" data-anchor-id="cell-culture-and-generation-of-stable-cell-lines">Cell culture and generation of stable cell lines</h2>
<p>Drosophila S2 cells were cultured at 27°C in Schneider’s medium (Thermo Fisher Scientific) supplemented with 10% fetal bovine serum (Omega Scientific) and antibiotic/antimycotic (Thermo Fisher Scientific). For the generation of stable cell lines, cells were plated at 5✕10<sup>5</sup> cells per mL. After 24 hours, cells were transfected with 10 µg plasmid DNA using Effectene transfection reagent (Qiagen). After an additional 24 hours, puromycin was added to a final concentration of 2 µg/mL. Stable cell lines were recovered after 2-3 weeks of selection. Following recovery of stable cell lines, cells were cultured with 1 µg/mL puromycin.</p>
</section>
<section id="induction-of-transcription-factor-expression" class="level2">
<h2 class="anchored" data-anchor-id="induction-of-transcription-factor-expression">Induction of transcription factor expression</h2>
<p>Transcription factor expression was induced by adding CuSO<sub>4</sub> to the cell culture media. Expression of protein at physiological levels was achieved with 1000, 100, or 40 µM CuSO<sub>4</sub> for Zld, Grh or Twi, respectively (Supplemental figure X). For experiments with sub- or supraphysiological protein levels (Fig. 5), the following CuSO<sub>4</sub> concentrations were used: 500 µM (Zld subphysiological), 1000 µM (Zld physiological) or 1500 µM (Zld supraphysiological); 25 µM (Grh subphysiological), 100 µM (Grh physiological) or 400 µM (Grh supraphysiological); 10 µM (Twi subphysiological), 40 µM (Twi physiological) or 160 µM (Twi supraphysiological). For all induction experiments, cells were plated at 1×10<sup>6</sup> cells per mL. Unless otherwise specified, experiments were performed with cells harvested 48 hours after induction.</p>
</section>
<section id="immunoblotting" class="level2">
<h2 class="anchored" data-anchor-id="immunoblotting">Immunoblotting</h2>
<p>Proteins were separated on denaturing polyacramide gels before transfer to a 0.45 μm Immobilon-P PVDF membrane (Millipore) in transfer buffer (25 mM Tris, 200 mM Glycine, 20% methanol) for 60 min (75 min for Zld) at 500mA at 4°C. The membranes were blocked with BLOTTO (2.5% non-fat dry milk, 0.5% BSA, 0.5% NP-40, in TBST) for 30 min at room temperature and then incubated with anti-Zld (1:750) (Harrison et al., 2010), anti-Grh (1:1000) (Harrison et al., 2010), anti-Twi (1:1000) (He et al.&nbsp;2015), anti-HA-peroxidase (1:500) (clone 3F10, Roche), or anti-Tubulin (DM1A, 1:5000) (Sigma), overnight at 4°C. The secondary incubation was performed with goat anti-rabbit IgG-HRP conjugate (1:3000) (Bio-Rad) for 1 hr at room temperature. Blots were treated with SuperSignal West Pico PLUS chemiluminescent substrate (Thermo Fisher Scientific) and visualized using the Azure Biosystems c600 or Kodak/Carestream BioMax Film (VWR).</p>
</section>
<section id="immunostaining" class="level2">
<h2 class="anchored" data-anchor-id="immunostaining">Immunostaining</h2>
</section>
<section id="chip-seq" class="level2">
<h2 class="anchored" data-anchor-id="chip-seq">ChIP-seq</h2>
<p>For each ChIP experiment, ~50 million cells were crosslinked by adding methanol-free formaldehyde directly to the cell culture medium to a final concentration of 0.8%. Cells were rotated on a nutator for 7 minutes at room temperature (RT) before quenching of crosslinking by adding glycine to 125 mM. Crosslinked cells were pelleted by centrifugation at 600 × <em>g</em> for 3 minutes. Cells were washed twice with 1X PBS. Cells were lysed by resuspending in lysis buffer (50 mM HEPES pH 7.9, 140 mM NaCl, 1 mM EDTA, 10% glycerol, 0.5% NP40, 0.25% Triton-X 100) with protease inhibitors (Pierce) and incubating on ice for 10 minutes. The resulting chromatin was pelleted for 5 minutes at 1500 × <em>g</em> and resuspended in RIPA buffer. Chromatin was sonicated in a Covaris S220 Ultrasonicator (5 cycles of 120 seconds with 60 second delay, 170 peak power, 10% duty factor, 200 cycles/burst). Sonicated chromatin was centrifuged for 10 minutes at 10,000 × <em>g</em> to pellet insoluble material. 5% of the supernatant was set aside as input, and the remainder was incubated at 4°C overnight with antibodies (6 µL anti-Zld, 8 µL anti-Grh, 10 µg anti-Twi, 7.5 µL anti-HA). 20 µL protein A beads (Dynabeads Protein A, ThermoFisher Scientific) blocked with BSA were added and samples were incubated at 4°C for 4 hours. Beads were separated on a magnet and washed 3× with low salt wash buffer (10 mM Tris pH 7.6, 1 mM EDTA, 0.1% SDS, 0.1% Na-Deoxycholate, 1% Triton-X 100, 150 mM NaCl), 2× high salt wash buffer (10 mM Tris pH 7.6, 1 mM EDTA, 0.1% SDS, 0.1% sodium deoxycholate, 1% Triton-X 100, 300 mM NaC), 2× LiCl wash buffer (0.25 M LiCl, 0.5% NP40, 0.5% sodium deoxycholate), and 1× with TE buffer with NaCl (10 mM Tris-HCl pH 8.0, 1 mM EDTA, 50 mM NaCl). Beads were resuspended in elution buffer (50 mM Tris-HCl pH 8.0, 1% SDS, 10 mM EDTA) and incubated at 65°C for 10 minutes. The IP and input samples were treated with 4.5 µL RNAse A for 30 minutes. 5 µL Proteinase K was added and samples were incubated overnight at 65°C to reverse crosslinking. DNA was isolated by phenol:chloroform extraction, precipitated with EtOH, and resuspended in 20 µL H<sub>2</sub>O.Preparation of sequencing libraries was performed using NEBNext Ultra II kit (NEB) with 7 PCR cycles for library amplification. Sequencing was performed on the Illumina Hi-Seq4000 using 50bp single-end reads, or on the Illumina NovaSeq 6000 using 150bp paired-end reads.</p>
</section>
<section id="chip-seq-analysis" class="level2">
<h2 class="anchored" data-anchor-id="chip-seq-analysis">ChIP-seq analysis</h2>
<p>Bowtie2 version 2.4.4 <span class="citation" data-cites="Langmead2012">(<a href="#ref-Langmead2012" role="doc-biblioref">Langmead and Salzberg 2012</a>)</span> was used to align ChIP-seq reads to the <em>Drosophila melanogaster</em> genome (version dm6) with the following non-default parameters: -k 2 --very-sensitive --no-mixed --no-discordant -X 5000. Aligned reads were filtered to include only reads with a mapping quality score &gt; 30. Reads mapping to unplaced scaffolds or the mitochondrial genome were discarded.</p>
<p>Alignment</p>
<p>Filtering</p>
<p>peak calling</p>
<p>peak filtering</p>
<p>generation of bigwigs</p>
<p>normalization of bigwigs</p>
</section>
<section id="rna-seq" class="level2">
<h2 class="anchored" data-anchor-id="rna-seq">RNA-seq</h2>
<p>5×10<sup>6</sup> cells were harvested and resuspended in 800 µL Trizol (Invitrogen). RNA was purified using a chloroform extraction followed by a column-based Quick-RNA MiniPrep extraction (Zymo). Poly-A-selected RNA sequencing (RNA-seq) libraries were prepared using the TruSeq RNA sample prep kit v2 (Illumina).</p>
</section>
<section id="rna-seq-analysis" class="level2">
<h2 class="anchored" data-anchor-id="rna-seq-analysis">RNA-seq analysis</h2>
<p>Raw reads were aligned to the Drosophila melanogaster (dm6) genome using HISAT2 version 2.1.0 with parameters -k 2 –very-sensitive. Reads with a mapping quality score &lt; 30 and reads aligning to the mitochondrial genome or unplaced scaffolds were discarded. FeatureCounts (from Subread version 2.0) was used to assign aligned reads to genes (FlyBase gene annotation release r6.45). The resultant table of read counts was used to perform differential expression analysis using DESeq2 <span class="citation" data-cites="love2014">(<a href="#ref-love2014" role="doc-biblioref">Love, Huber, and Anders 2014</a>)</span>. Genes with an adjusted p-value &lt; 0.05 and a fold change &gt; 2 were considered differentially expressed.</p>
</section>
<section id="atac-seq" class="level2">
<h2 class="anchored" data-anchor-id="atac-seq">ATAC-seq</h2>
</section>
<section id="atac-seq-analysis" class="level2">
<h2 class="anchored" data-anchor-id="atac-seq-analysis">ATAC-seq analysis</h2>
</section>
<section id="cutrun" class="level2">
<h2 class="anchored" data-anchor-id="cutrun">CUT&amp;RUN</h2>
</section>
<section id="cutrun-analysis" class="level2">
<h2 class="anchored" data-anchor-id="cutrun-analysis">CUT&amp;RUN analysis</h2>
</section>
<section id="treatment-of-s2-cells-with-tazemetostat" class="level2">
<h2 class="anchored" data-anchor-id="treatment-of-s2-cells-with-tazemetostat">Treatment of S2 cells with tazemetostat</h2>
</section>
<section id="analysis-of-previously-published-data" class="level2">
<h2 class="anchored" data-anchor-id="analysis-of-previously-published-data">Analysis of previously published data</h2>
</section>
<section id="data-and-code-availability" class="level2">
<h2 class="anchored" data-anchor-id="data-and-code-availability">Data and code availability</h2>
</section>
</section>
<section id="figure-legends" class="level1">
<h1>Figure Legends</h1>
<p><strong>Figure 1: Ectopic pioneer factors open chromatin and activate transcription.</strong></p>
<p><strong>Figure 2: Twist binds closed chromatin extensively and drives accessibility at a limited number of sites.</strong></p>
<p><strong>Figure 3: Motif content shapes pioneer-factor activity</strong></p>
<p><strong>Figure 4: H3K27me3 and H3K9me3 are barriers to pioneer-factor binding.</strong></p>
<p><strong>Figure 5: Binding and opening of closed chromatin is concentration dependent.</strong></p>
<p><strong>Figure 6: Regions outside the DNA-binding domain are required for pioneer-factor function.</strong></p>
<p><strong>Figure 7: Pioneer-factor function requires stable chromatin occupancy.</strong></p>
<p><strong>Extended Data Figure 1: Stable cell lines allow inducible expression of transcription factors at physiological concentrations.</strong></p>
<ul>
<li><p>Schematic of cell line generation.</p></li>
<li><p>RNA-seq data to show expression of S2 and not Kc cell markers.</p></li>
<li><p>Western blots to determine physiological expression and lack of protein expression in WT cells.</p></li>
<li><p>RNA-seq data to show leaky expression and strong induction of RNA levels.</p></li>
</ul>
<p><strong>Extended Data Figure 2: Ectopically expressed Zld and Grh bind to thousands of sites in S2 cells.</strong></p>
<ul>
<li><p>ChIP-seq replicate correlation</p></li>
<li><p>Genome browser and heatmaps for all ChIP peaks?</p></li>
<li><p>IgG and WT controls for antibody specificity</p></li>
</ul>
<p><strong>Extended Data Figure 3: Ectopic expression of pioneer factors in S2 cells leads to widespread changes to chromatin accessibility.</strong></p>
<ul>
<li><p>ATAC PCA plot</p></li>
<li><p>ATAC replicate correlation</p></li>
<li><p>volcano and/or MA plots with enrichment of binding</p></li>
<li><p>Example genome browsers?</p></li>
<li><p>Analysis of nonspecific changes?</p></li>
</ul>
<p><strong>Extended Data Figure 4: Zld and Grh bind to chromatin rapidly after induction of protein expression</strong></p>
<ul>
<li><p>Western blot time course of protein levels</p></li>
<li><p>CUT&amp;RUN time course to show rapid binding</p></li>
<li><p>likely rules out dependence on DNA replication</p></li>
</ul>
<p><strong>Extended Data Figure 5: Zld and Grh binding promotes transcriptional activation at a subset of binding sites</strong></p>
<ul>
<li><p>RNA-seq PCA</p></li>
<li><p>RNA-seq replicate correlation</p></li>
<li><p>RNA-seq volcano/MA plots</p></li>
<li><p>RNA-seq GO analysis</p></li>
</ul>
<p><strong>Information to include in supplemental figures:</strong></p>
<ul>
<li><p>De novo motif results</p></li>
<li><p>Analysis of promoter motifs</p></li>
</ul>

</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-Cherbas2011" class="csl-entry" role="doc-biblioentry">
Cherbas, Lucy, Aarron Willingham, Dayu Zhang, Li Yang, Yi Zou, Brian D. Eads, Joseph W. Carlson, et al. 2010. <span>“The Transcriptional Diversity of 25 <span><em>Drosophila</em></span> Cell Lines.”</span> <em>Genome Research</em> 21 (2): 301–14. <a href="https://doi.org/10.1101/gr.112961.110">https://doi.org/10.1101/gr.112961.110</a>.
</div>
<div id="ref-Langmead2012" class="csl-entry" role="doc-biblioentry">
Langmead, Ben, and Steven L Salzberg. 2012. <span>“Fast Gapped-Read Alignment with Bowtie 2.”</span> <em>Nature Methods</em> 9 (4): 357–59. <a href="https://doi.org/10.1038/nmeth.1923">https://doi.org/10.1038/nmeth.1923</a>.
</div>
<div id="ref-love2014" class="csl-entry" role="doc-biblioentry">
Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. <span>“Moderated Estimation of Fold Change and Dispersion for RNA-Seq Data with DESeq2.”</span> <em>Genome Biology</em> 15 (12). <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.
</div>
<div id="ref-Sandmann2007" class="csl-entry" role="doc-biblioentry">
Sandmann, Thomas, Charles Girardot, Marc Brehme, Waraporn Tongprasit, Viktor Stolc, and Eileen E. M. Furlong. 2007. <span>“A Core Transcriptional Network for Early Mesoderm Development in <span><em>Drosophila Melanogaster</em></span>.”</span> <em>Genes &amp; Development</em> 21 (4): 436–49. <a href="https://doi.org/10.1101/gad.1509007">https://doi.org/10.1101/gad.1509007</a>.
</div>
<div id="ref-Wilczynski2010a" class="csl-entry" role="doc-biblioentry">
Wilczyński, Bartek, and Eileen E M Furlong. 2010. <span>“Dynamic CRM Occupancy Reflects a Temporal Map of Developmental Progression.”</span> <em>Molecular Systems Biology</em> 6 (1): 383. <a href="https://doi.org/10.1038/msb.2010.35">https://doi.org/10.1038/msb.2010.35</a>.
</div>
<div id="ref-Yáñez-Cuna2012" class="csl-entry" role="doc-biblioentry">
Yáñez-Cuna, J. Omar, Huy Q. Dinh, Evgeny Z. Kvon, Daria Shlyueva, and Alexander Stark. 2012. <span>“Uncovering <span><em>Cis</em></span>-Regulatory Sequence Requirements for Context-Specific Transcription Factor Binding.”</span> <em>Genome Research</em> 22 (10): 2018–30. <a href="https://doi.org/10.1101/gr.132811.111">https://doi.org/10.1101/gr.132811.111</a>.
</div>
</div></section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>